site/app/views/forum/ForumThreadView.php

```diff
@@ -1096,8 +1100,25 @@ public function createPost(array $thread, array $post, $unviewed_posts, $first,
         }
 
         if ($post["anonymous"]) {
-            $visible_username = "Anonymous";
+            if($first){
+                $visible_username = "Anonymous";
+            }
+            else if($post["author_user_id"] == $first_post_start['author_user_id']){
+                $visible_username = "Anonymous [OP]";
+            }
+            else{
+                $visible_username = "Anonymous";
+            }
         }
+        else if(!$post["anonymous"]){
+            if($first){
+                $visible_username = $visible_username . " [OP]";
+            }
+            else if($post["author_user_id"] == $first_post_start['author_user_id']){
```

Comment:
In regards to the backend code, it might be simplest just to check if anonymous and set $visible_username like the main branch does, and then check if it's OP's post and concatenate the " [OP]" tag afterwards.